pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionCommon.java

```diff
@@ -337,6 +338,16 @@ public static final long getSequenceId(MessageId messageId) {
         return offset;
     }
 
+    public static final long getSequenceId(Message message) {
+        // Use index added by org.apache.pulsar.common.intercept.AppendIndexMetadataInterceptor if present.
+        // Requires exposingBrokerEntryMetadataToClientEnabled=true on brokers.
+        if (message.getIndex().isPresent()) {
+            return (long) message.getIndex().get();
```

Comment:
The `AppendIndexMetadataInterceptor` uses a self-incrementing count as the index, and its value may conflict with other topics, causing problems for any function with multiple input topics. A better solution might be to add the batch index into the `getSequenceId(MessageId messageId)`. WDYT?

Reply:
Thanks @freeznet . Add batch index is our first idea, but we found that ledgerId is used up quickly. In fact, KOP has encountered the same problem before #9038 , this way makes the max range of ledgerId much lower than Long.MAX_VALUE.  
In #9039 , each partition will be assigned a ManagedLedgerInterceptor when ManagedLedger created; Each Entry will be intercept for adding a monotone increasing offset in Broker entry metadata and the offet is added by batchSize of entry; Therefore,  its value will not conflict with other topic. I have test it in multiple topics and partitions and it work great